components/edit_channel_header_modal/edit_channel_header_modal.tsx

```diff
@@ -100,21 +100,24 @@ export default class EditChannelHeaderModal extends React.PureComponent<Props, S
     }
 
     public handleSave = async (): Promise<void> => {
-        const {header} = this.state;
-        if (header === this.props.channel.header) {
-            this.hideModal();
-            return;
+        const header = this.cleanHeaderText();
+
+        if (header != this.props.channel.header) {
+            this.setState({saving: true});
+            const {channel, actions} = this.props;
+            const {error} = await actions.patchChannel(channel.id!, {header});
+            if (error)
+                this.setState({serverError: error, saving: false});
+                console.log(error)
         }
+        console.log(this.state)
+        this.hideModal()
+    }
 
-        this.setState({saving: true});
-
-        const {channel, actions} = this.props;
-        const {error} = await actions.patchChannel(channel.id!, {header});
-        if (error) {
-            this.setState({serverError: error, saving: false});
-        } else {
-            this.hideModal();
-        }
+    private cleanHeaderText = () => {
+        let {header} = this.state
+        header = header?.trim()
+        return header
```

Comment:
Do we need a separate function ? it looks simple enough to me to be written in a single line